### PR TITLE
[browser] Remove WASM `HybridGlobalization` from runtime code

### LINKED ISSUE #102373: [browser][hybridglobalization][aot] `Assert.StartsWith` cannot find a match
A lot of failures in `_HybridGlobalization_AOT` lane, scenario: `WasmTestOnV8` and `WasmTestOnChrome`

examples:
```
[FAIL] Microsoft.Extensions.Configuration.Xml.Test.XmlConfigurationExtensionsTest.AddXmlFile_ThrowsIfFileDoesNotExistAtPath
info: Assert.StartsWith() Failure: String start does not match
 info: String:         "The configuration file 'NotExistingConfig"���
 info: Expected start: "The configuration file 'NotExistingConfig"���
info:    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object , BindingFlags )
info: [FAIL] Microsoft.Extensions.Configuration.Xml.Test.XmlConfigurationTest.XmlConfiguration_Throws_On_Missing_Configuration_File
info: Assert.StartsWith() Failure: String start does not match
info: String:         "The configuration file 'NotExistingConfig"���
info: Expected start: "The configuration file 'NotExistingConfig"���
info:    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object , BindingFlags )
 info: Finished:    Microsoft.Extensions.Configuration.Xml.Tests.dll
```

```
[FAIL] Microsoft.Extensions.Configuration.Xml.Test.XmlConfigurationTest.XmlConfiguration_Throws_On_Missing_Configuration_File
info: Assert.StartsWith() Failure: String start does not match
info: String:         "The configuration file 'NotExistingConfig"···
info: Expected start: "The configuration file 'NotExistingConfig"···
info:    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object , BindingFlags )
info: [FAIL] Microsoft.Extensions.Configuration.Xml.Test.XmlConfigurationExtensionsTest.AddXmlFile_ThrowsIfFileDoesNotExistAtPath
info: Assert.StartsWith() Failure: String start does not match
info: String:         "The configuration file 'NotExistingConfig"···
info: Expected start: "The configuration file 'NotExistingConfig"···
info:    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object , BindingFlags )
info: Finished:    Microsoft.Extensions.Configuration.Xml.Tests.dll
```

### PR DESCRIPTION
**Pull request by ilonatommy**
Connected PRs:
https://github.com/dotnet/runtime/pull/110526
https://github.com/dotnet/runtime/pull/110534

This is the last PR aiming at removing `HybridGlobalization` on WASM. Removals:
- code from JS, including removal of hybrid globalization module from rollup
- code from icalls
- code from libraries, including stubs
- passing the flag
- processing the hybrid globalization assets, including segmentation-rules
- removing hybrid-specific asserts in WBT
- removing Blazor hybrid WBT

Renames / preserved code:
We are leaving JS-based globalization method `GetLocaleInfo` to support APIs like `NativeName` and `DisplayName` in the regular, ICU globalization mode. The code will be exported by `src\mono\browser\runtime\globalization.ts`.
- rename `locales-common.ts` -> `globalization-locale.ts`
- move useful helper methods to `globalization-locale.ts`

Closes https://github.com/dotnet/runtime/issues/102373, https://github.com/dotnet/runtime/issues/95795, https://github.com/dotnet/runtime/issues/98483, https://github.com/dotnet/runtime/issues/95921

### CODE REVIEW

**Comment by ilonatommy** [comment]
We finish with
```
nugets\microsoft.dotnet.sharedframework.sdk\10.0.0-beta.24605.1\targets\sharedfx.targets(305,5): error : The following files are missing entries in the templated manifest: icudt_hybrid.dat. Add these file names with extensions to the 'PlatformManifestFileEntry' item group for the runtime pack and corresponding ref pack to include them in the platform manifest.
```
It seems sdk change is needed first, then this PR.

**Comment by ilonatommy** [comment]
WBT tests will fail until https://github.com/dotnet/icu/pull/536 flows in. From this reason, I am reverting WBT changes as well. They will be introduced in another PR. Removals for the follow up:
- https://github.com/dotnet/runtime/blob/fc620a3525288ac0dff1ef007996e9575e7c1be9/src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props#L169
- https://github.com/dotnet/runtime/blob/fc620a3525288ac0dff1ef007996e9575e7c1be9/src/mono/wasm/Wasm.Build.Tests/ProjectProviderBase.cs#L363C1-L365C57

**Review by javiercn — APPROVED** [review]
Without having all the context, the changes look good, and I imagine tests will pass

File: src/mono/browser/runtime/exports.ts [hunk 1]
```diff
@@ -42,22 +40,12 @@ function initializeExports (globalObjects: GlobalObjects): RuntimeAPI {
...
         forceDisposeProxies,
-        instantiate_segmentation_rules_asset,
 
     };
     if (WasmEnableThreads) {
         rh.dumpThreads = mono_wasm_dump_threads;
         rh.mono_wasm_print_thread_dump = () => tcwraps.mono_wasm_print_thread_dump();
     }
-    if (loaderHelpers.config.globalizationMode === GlobalizationMode.Hybrid) {
-        rh.stringToUTF16 = stringToUTF16;
```
Comment (pavelsavara) [inline src/mono/browser/runtime/exports.ts:53]: is any of the string functions now un-used ?

**Review by pavelsavara — APPROVED** [review]
I think `instantiate_segmentation_rules_asset` should be deleted too.

I wonder if this will break any file-based SDK tests

**Review by radekdoulik — APPROVED** [review]
LGTM

File: src/mono/browser/runtime/exports.ts (hunk 1 above)
Comment (ilonatommy) [reply src/mono/browser/runtime/exports.ts:53]: `instantiate_segmentation_rules_asset` is deleted from `src/mono/browser/runtime/assets.ts`, I cannot find any other occurrence that is left, so I guess it's okay.
`stringToUTF16` is used but it's taken from the main module. This one that is deleted in the quoted code is a forwarded method to hybrid globalization module. During removal of hybrid code, we also removed the hybrid module, that's why this line is fine.

Comment (ilonatommy) [reply src/mono/browser/runtime/exports.ts:53]: "file-based SDK tests" what is a file-based test?

**Review by akoeplinger — APPROVED** [review]

**Comment by akoeplinger** [comment]
> WBT tests will fail until https://github.com/dotnet/icu/pull/536 flows in.

@ilonatommy you can run this command locally to get the new ICU:
```
darc update-dependencies --source-repo icu --channel ".NET 10"
```

File: src/mono/browser/runtime/exports.ts (hunk 1 above)
Comment (pavelsavara) [reply src/mono/browser/runtime/exports.ts:53]: but for .js file

**Comment by ilonatommy** [comment]
> @ilonatommy you can run this command locally to get the new ICU:
> 
> ```
> darc update-dependencies --source-repo icu --channel ".NET 10"
> ```

{F4367A02-5808-4D6C-9B55-348A0FBE3980}

**Comment by akoeplinger** [comment]
@ilonatommy you need to run it in a checkout of this PR

**Comment by akoeplinger** [comment]
> Which means it's only a local change, it does not make a maestro PR to update main?

Correct, you'd need to commit it like any other change.

**Pull request merged**